### DISCUSSION TITLE
My Argocd Failed to synchronize application status. "Failed to load target state: failed to generate manifest for source 1 of 1: rpc error: code = Unknown desc = unknown error: remote: Internal API unreachable"

### DISCUSSION THREAD

**Discussion by pantao24** [argoproj/argo-cd#17608] [Q&A]
![image-20240321203157802](https://github.com/argoproj/argo-cd/assets/40918197/bcf1c2e6-4598-4adc-b096-11c6c25a9a28)

**Comment by jgwest** [comment]
This sounds like the Argo CD Application controller component is not able to connect to the Argo CD reposerver component. You should verify that both are healthy, and application controller has required network permissions.

**Comment by Nikhil007nsg** [comment]
can u tell me steps how to resolve this , i allready created secret and configmap file of github credentials but still this problem are there

**Reply by hemantharjunagari** [reply]
is it solved @Nikhil007nsg, if yes can you tell the steps how to debug this? Thanks in advance

**Comment by chekkan** [comment]
Can you post what you see in the logs?

```
kubectl logs -n argocd -l app.kubernetes.io/name=argocd-application-controller
```

I faced the same issue and saw this message...

```
time="2025-12-11T12:58:33Z" level=error msg="Failed to cache app resources: error setting app resource tree: NOAUTH Authentication required." app-namespace=argocd application=kargo comparison-level=2 dedup_ms=2 dest-name= dest-namespace=kargo dest-server="https://kubernetes.default.svc" diff_ms=394 git_ms=740 health_ms=12 live_ms=4 project=infra settings_ms=0 sync_ms=0
```

The problem was that I had the config to skip `redisSecretInit`.

I re-enables Redis secret initialisation in my `values.yaml`:

```yaml
argo-cd:
  redisSecretInit:
    enabled: true
```

Then manually triggered sync

```
kubectl patch application argocd -n argocd --type merge -p '{"operation":{"sync":{}}}'
```

Wait for the init job to complete:

```
kubectl get job argocd-redis-secret-init -n argocd
```

Restart ArgoCD components to pick up the new Redis password:

```
kubectl rollout restart deployment/argocd-server -n argocd
kubectl rollout restart deployment/argocd-repo-server -n argocd
kubectl rollout restart statefulset/argocd-application-controller -n argocd
kubectl rollout restart deployment/argocd-redis -n argocd
```

After following these steps, all my applications transitioned from `ComparisonError` to `Synced` status successfully.
